### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by solon-yang-ck**
removes good-http and good-udp. They're both deprecated and we only use simple code from them. I've moved the associated code to this codebase and updated the HAPI dependencies.

All Tests pass, exactly 95% coverage.

### CODE REVIEW

File: lib/good-influx-http.js [hunk 1]
```diff
@@ -28,11 +67,14 @@ class GoodInfluxHttp extends GoodHttp {
             payload: lineData.join('\n')
         };
 
-        Hoek.merge(wreckOptions, this._settings.wreck, false);
+        Hoek.merge(wreckOptions, this._settings.wreck, { nullOverride: false });
 
         // Prevent this from user tampering
-        wreckOptions.headers['content-type'] = 'text/plain';
-        Wreck.request('post', this._endpoint, wreckOptions, callback);
+        wreckOptions.headers ? wreckOptions.headers['content-type'] = 'text/plain'
```
Comment (solon-yang-ck) [inline lib/good-influx-http.js]: change was needed so this error wouldn't occur:
```
26) GoodInflux HTTP URL => Sends events in a stream to HTTP server:

      Cannot set property 'content-type' of undefined

      at GoodInfluxHttp._sendMessages (/Users/solon.yang/good-influx/lib/good-influx-http.js:73:130)
      at GoodInfluxHttp._write (/Users/solon.yang/good-influx/lib/good-influx-http.js:42:102)
      at doWrite (_stream_writable.js:397:12)
      at clearBuffer (_stream_writable.js:524:7)
      at onwrite (_stream_writable.js:449:7)
      at runCallback (timers.js:810:20)
      at tryOnImmediate (timers.js:768:5)
      at processImmediate [as _immediateCallback] (timers.js:745:5)
```

File: lib/good-influx-http.js [hunk 2]
```diff
@@ -28,11 +67,14 @@ class GoodInfluxHttp extends GoodHttp {
...
+            : wreckOptions.headers = { 'content-type': 'text/plain' };
+        Wreck.request('post', this._endpoint, wreckOptions).then((req) => {
```
Comment (solon-yang-ck) [inline lib/good-influx-http.js:88]: New Wreck doesn't allow for callback, so instead, I duplicated the old wreck behavior.

**Review by OuranosSkia — APPROVED** [review]

File: lib/good-influx-http.js [hunk 3]
```diff
@@ -1,23 +1,70 @@
...
     }
 };
 
-class GoodInfluxHttp extends GoodHttp {
+class GoodInfluxHttp extends Stream.Writable {
     constructor(endpoint, config, schemas) {
-        super(endpoint, config, schemas);
-        this._settings.schema = 'good-influx';
+        config = config || {};
+        const settings = Object.assign({}, internals.defaults, config);
```
Comment (bradley-marker-ck) [inline lib/good-influx-http.js:27]: I prefer the object spread and it actually supports undefined (and I think null) so you wouldn't have to default `config`.

File: lib/good-influx-http.js [hunk 4]
```diff
@@ -1,23 +1,70 @@
...
+
+        if (settings.errorThreshold === null) {
+            settings.errorThreshold = -Infinity;
+        }
+
+        super({ objectMode: true, decodeStrings: false });
```
Comment (bradley-marker-ck) [inline lib/good-influx-http.js]: This does not seem to depend on the code above it. Seems like you could move the `super` to the first line of the constructor.

File: lib/good-influx-http.js [hunk 5]
```diff
@@ -1,23 +1,70 @@
...
+
+        // Standard users
+        this.once('finish', () => {
+
+            this._sendMessages();
+        });
         this.config = Hoek.applyToDefaults(internals.defaults, config);
         this.schemas = schemas;
     }
+    _write(data, encoding, callback) {
```
Comment (bradley-marker-ck) [inline lib/good-influx-http.js:47]: nit: Maybe it's just the git diff but I don't see a new-line between this function and the constructor.

File: lib/good-influx-http.js (hunk 1 above)
Comment (bradley-marker-ck) [reply lib/good-influx-http.js]: Ternary assignment looks really odd to me. I'd actually rather see an if/else for an assignment.

The point of a ternary is usually get get the result.

File: lib/good-influx-udp.js [hunk 6]
```diff
@@ -1,26 +1,61 @@
...
+
+        // Standard users
+        this.once('finish', () => {
+
+            this._sendMessages();
+        });
         this.config = Hoek.applyToDefaults(internals.defaults, config);
         this.schemas = schemas;
     }
+    _write(data, encoding, callback) {
```
Comment (bradley-marker-ck) [inline lib/good-influx-udp.js:47]: nit: New-line again or github diff being unclear.

File: lib/good-influx-http.js (hunk 4 above)
Comment (solon-yang-ck) [reply lib/good-influx-http.js]: fixed

File: lib/good-influx-http.js (hunk 3 above)
Comment (solon-yang-ck) [reply lib/good-influx-http.js:27]: can't do that. the eslint settings don't allow for the spread operator.

Comment (bradley-marker-ck) [reply lib/good-influx-http.js:27]: :(

File: package.json [hunk 7]
```diff
@@ -11,14 +11,13 @@
...
     "fast-safe-stringify": "^2.0.2",
-    "good-http": "^6.1.3",
-    "good-udp": "4.1.x",
-    "hoek": "^4.1.1",
-    "lodash": "^4.17.10",
-    "wreck": "^12.2.0"
+    "lodash": "^4.17.10"
   },
   "devDependencies": {
+    "@types/node": "^13.1.7",
```
Comment (bradley-marker-ck) [inline package.json]: Node 13? Should be 8 or 10 I'd think ... since we still have some Node 8 services, I think 8.

**Review by bradley-marker-ck — APPROVED** [review]

**Pull request merged**